nrf_security/src/psa_crypto_driver_wrappers.c

```diff
@@ -97,6 +98,216 @@
+                                     size_t bits,
+                                     psa_algorithm_t algorithm)
+{
+    switch (curve)
+    {
+    case PSA_ECC_FAMILY_SECP_K1:
+        if( is_ecdsa_or_ecdh(algorithm) &&
+            (bits == 192 || bits == 224 || bits == 256)){
+            return PSA_SUCCESS;
+        }
+        break;
+
+    case PSA_ECC_FAMILY_SECP_R1:
+        if(is_ecdsa_or_ecdh(algorithm) &&
+           (bits == 192 || bits == 224 ||
+            bits == 256 || bits == 384 || bits == 521)){
+            return PSA_SUCCESS;
+        }
+        break;
+
+    case PSA_ECC_FAMILY_SECT_K1:
+        if(is_ecdsa_or_ecdh(algorithm) &&
+           (bits == 163 || bits == 233 || bits == 239 ||
+            bits == 283 || bits == 409 || bits == 571)){
+            return PSA_SUCCESS;
+        }
+        break;
+
+    case PSA_ECC_FAMILY_SECT_R1:
+        if(is_ecdsa_or_ecdh(algorithm) &&
+           (bits == 163 || bits == 233 || 
+            bits == 283 || bits == 409 || bits == 571)){
+            return PSA_SUCCESS;
+        }
+        break;
+
+    case PSA_ECC_FAMILY_SECT_R2:
+        if(is_ecdsa_or_ecdh(algorithm) && bits == 163){
+            return PSA_SUCCESS;
+        }
+        break;
+
+    case PSA_ECC_FAMILY_BRAINPOOL_P_R1:
+        if(is_ecdsa_or_ecdh(algorithm) &&
+           (bits == 160 || bits == 192 || bits == 224 ||
+            bits == 256 || bits == 320 || bits == 384 || bits == 512)){
+            return PSA_SUCCESS;
+        }
+        break;
+
+    case PSA_ECC_FAMILY_MONTGOMERY:
+        if(algorithm == PSA_ALG_ECDH && 
+           (bits == 255 || bits == 448)){ 
+            return PSA_SUCCESS;
+        } 
+        break;
+    
+    case PSA_ECC_FAMILY_TWISTED_EDWARDS:
+        if((algorithm == PSA_ALG_PURE_EDDSA ||
+           algorithm == PSA_ALG_ED25519PH ||    // (Edwards25519 only)
+           algorithm == PSA_ALG_ED448PH) &&     // (Edwards448 only)
+           (bits == 255 || bits == 448)){ 
+            return PSA_SUCCESS;
+        }
+        break;
+
+    default:
+        return( PSA_ERROR_INVALID_ARGUMENT );
+        break;
+    }
+
+    return PSA_ERROR_INVALID_ARGUMENT;
+}
+
+/**
+ * @brief Checks if the given combination of key type, bit size and algorithm 
+ *  is a valid tuple by the PSA crypto spec. 
+ *  
+ * @return psa_status_t psa_status_t PSA_ERROR_INVALID_ARGUMENT if invalid else PSA_SUCCESS
+ */
+static psa_status_t validate_key(psa_key_type_t type, 
+                                 size_t bits, 
+                                 psa_algorithm_t algorithm)
+{
+    if (PSA_KEY_TYPE_IS_ECC(type)){
+        psa_ecc_family_t curve = PSA_KEY_TYPE_ECC_GET_FAMILY(type);
+        return validate_ecc_key(curve, bits, algorithm);
+    }
+
+    if( bits % 8 != 0 ) {
+        return( PSA_ERROR_INVALID_ARGUMENT );
+    }
+
+    switch( type )
+    {
+        case PSA_KEY_TYPE_RAW_DATA:
+        case PSA_KEY_TYPE_DERIVE:
+            if(GET_ALG_BASE(algorithm) == PSA_ALG_HKDF_BASE ||
```

Comment:
There is a macro for and for the TLS12 functions.
You can grep "PSA_ALG_IS" in the include/psa/ and you will see all the available macros.